src/game/N3UIDebug.cpp

```diff
@@ -0,0 +1,209 @@
+    style.Colors[ImGuiCol_FrameBg] = ImVec4(0.22f, 0.32f, 0.69f, 0.54f);
+    style.Colors[ImGuiCol_TitleBg] = ImVec4(0.07f, 0.08f, 0.20f, 1.00f);
+    style.Colors[ImGuiCol_TitleBgActive] = ImVec4(0.17f, 0.19f, 0.40f, 1.00f);
+    style.Colors[ImGuiCol_Separator] = ImVec4(0.25f, 0.90f, 0.52f, 0.50f);
+    style.Colors[ImGuiCol_ResizeGrip] = ImVec4(0.16f, 0.91f, 0.27f, 0.20f);
+
+    return true;
+}
+
+void CN3UIDebug::BeginScene() {}
+
+void CN3UIDebug::EndScene() {
+    if (s_bReleaseCalled) {
+        return;
+    }
+
+    ImGui_ImplDX9_NewFrame();
+    ImGui_ImplWin32_NewFrame();
+    ImGui::NewFrame();
+
+    RenderDockingSpace();
+
+    static bool s_bDemoWindow = true;
+    ImGui::ShowDemoWindow(&s_bDemoWindow);
+
+    RenderGameMetrics();
+
+    ImGui::EndFrame();
+}
+
+void CN3UIDebug::Present() {
+    if (s_bReleaseCalled) {
+        return;
+    }
+
+    ImGui::Render();
+    ImGui_ImplDX9_RenderDrawData(ImGui::GetDrawData());
+
+    // Update and Render additional Platform Windows
+    if (ImGui::GetIO().ConfigFlags & ImGuiConfigFlags_ViewportsEnable) {
+        ImGui::UpdatePlatformWindows();
+        ImGui::RenderPlatformWindowsDefault();
+    }
+}
+
+extern IMGUI_IMPL_API LRESULT ImGui_ImplWin32_WndProcHandler(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam);
+
+LRESULT CN3UIDebug::WndProcMain(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam) {
+    if (ImGui_ImplWin32_WndProcHandler(hWnd, message, wParam, lParam)) {
+        return true;
+    }
+
+    if (ImGui::GetCurrentContext() && (ImGui::GetIO().WantCaptureMouse || ImGui::GetIO().WantCaptureKeyboard)) {
+        return true;
+    }
+
+    return false;
+}
+
+void CN3UIDebug::RenderDockingSpace() {
+    static ImGuiWindowFlags s_DockWindowFlags = ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoBackground |
+                                                ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoCollapse |
+                                                ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove |
+                                                ImGuiWindowFlags_NoBringToFrontOnFocus | ImGuiWindowFlags_NoNavFocus;
+
+    // Make sure the docking window fully covers the whole region.
+    ImGuiViewport * pVP = ImGui::GetMainViewport();
+    ImGui::SetNextWindowPos(pVP->Pos);
+    ImGui::SetNextWindowSize(pVP->Size);
+    ImGui::SetNextWindowViewport(pVP->ID);
+
+    // Foreground window style: we don't need it visible, so disable some props.
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowBorderSize, 0.0f);
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowPadding, ImVec2(0.0f, 0.0f));
+
+    // Begin
+    ImGui::Begin("Dockspace", NULL, s_DockWindowFlags);
+    ImGui::PopStyleVar();
+    ImGui::PopStyleVar(2);
+
+    ImGui::DockSpace(ImGui::GetID("MyDockspace"), ImVec2(0.0f, 0.0f), ImGuiDockNodeFlags_PassthruCentralNode);
+
+    ImGui::End();
+}
```

Comment:
Great point! Instead of removing it, I rewrote it to actually create the dockspace if no user persistent data exists: https://github.com/ko4life-net/ko/pull/168/commits/a4e91767725b7abf87f850cbe8ed0726c09fd3e8

Reply:
Nice work 🚀